### ISSUE TITLE
SSE Recovery Fails

### ISSUE THREAD

**Issue by SattaiLanfear** [akka/alpakka#206]
The tests for the SSE connector are missing a case where it fails.  Unfortunately while I tried to create a new test case, I'm not familiar enough with Akka HTTP serving to properly simulate the condition.

Problem:
The current tests only apply to a properly shut down interrupted stream.  The stream always properly closes because Akka is gracefully shutting down.  This fails to simulate actually losing a connection, or the remote server freezing.

In the case where an actual error is triggered by the incoming TCP stream, EventSource does not recover from the error and retry, it allows the whole stream to die.

Eg:

```console
akka.http.scaladsl.model.EntityStreamException: Entity stream truncation
	at akka.http.impl.engine.parsing.HttpMessageParser$$anonfun$chunkedEntity$1$$anonfun$2.applyOrElse(HttpMessageParser.scala:331)
	at akka.http.impl.engine.parsing.HttpMessageParser$$anonfun$chunkedEntity$1$$anonfun$2.applyOrElse(HttpMessageParser.scala:329)
	at akka.stream.impl.fusing.Collect$$anon$2$$anonfun$10.apply(Ops.scala:208)
	at akka.stream.impl.fusing.SupervisedGraphStageLogic.withSupervision(Ops.scala:169)
	at akka.stream.impl.fusing.Collect$$anon$2.onPush(Ops.scala:210)
	at akka.stream.impl.fusing.GraphInterpreter.processPush(GraphInterpreter.scala:747)
	at akka.stream.impl.fusing.GraphInterpreter.processEvent(GraphInterpreter.scala:710)
	at akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:616)
	at akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:471)
	at akka.stream.impl.fusing.GraphInterpreterShell.receive(ActorGraphInterpreter.scala:423)
	at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:603)
	at akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:618)
	at akka.actor.Actor$class.aroundReceive(Actor.scala:497)
	at akka.stream.impl.fusing.ActorGraphInterpreter.aroundReceive(ActorGraphInterpreter.scala:529)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:526)
	at akka.actor.ActorCell.invoke(ActorCell.scala:495)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:257)
	at akka.dispatch.Mailbox.run(Mailbox.scala:224)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:234)
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```

Note this isn't a failure at connecition time but the remote server simply vanishing.  I simulated this locally running a play server providing an event stream and connecting to it with a client running the EventSource.  When the play server was killed, the event stream died.

**Comment by hseeberger** [comment]
Good catch! So how could we write a test for that scenario which is really important?

**Comment by SattaiLanfear** [comment]
@hseeberger 
Maybe something could be done on the client side to simulate it since killing the server is impractical.

`akka.stream.UniqueKillSwitch` and the other KillSwitch variants allow triggering an arbitrary exception.  Unfortunately I don't think it could be put in to the stream at the right point to work, as any exception after the EventStream logic should be user-space.  Meanwhile even throwing an exception in the server side still leaves Akka cleaning up the socket properly and sending the final 0-size chunk.

**Comment by raboof** [comment]
Is this still a bug, or is the bug fixed with #275 (and does the wish for more tests remain)?

**Comment by hseeberger** [comment]
It seems to be fixed (somehow, not necessarily by #275).
